applications/FluidDynamicsApplication/custom_utilities/embedded_data.h

```diff
@@ -0,0 +1,104 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Jordi Cotela
+//
+
+#if !defined(KRATOS_EMBEDDED_DATA_H)
+#define KRATOS_EMBEDDED_DATA_H
+
+#include "fluid_dynamics_application_variables.h"
+#include "custom_utilities/fluid_element_data.h"
+
+namespace Kratos {
+
+///@addtogroup FluidDynamicsApplication
+///@{
+
+///@name Kratos classes
+///@{
+
+template< class TFluidData >
+class EmbeddedData : public TFluidData
+{
+public:
+
+///@name Type Definitions
+///@{
+
+using NodalScalarData = typename TFluidData::NodalScalarData;
+using NodalVectorData = typename TFluidData::NodalVectorData;
+
+typedef GeometryData::ShapeFunctionsGradientsType ShapeFunctionsGradientsType;
+typedef std::vector< Vector > InterfaceNormalsType;
+
+///@}
+///@name Public Members
+///@{
+
+NodalScalarData Distance;
+
+Matrix PositiveSideN;
```

Comment:
In the future we will need to add the negative side variables for the Ausas formulation. Maybe we can think in add them now.

Reply:
I thought about it too, but for now I didn't because I don't know where to use them. If the Ausas formulation fits within the same EmbeddedElement class, I'll add it, otherwise we'll have an EmbeddedAusasElement class with its EmbeddedAusasData.